### ISSUE TITLE
Renew update center certificate (crawler and update-center)

### ISSUE THREAD

**Issue by smerle33** [jenkins-infra/helpdesk#3489]
### Service(s)

trusted.ci.jenkins.io, Update center

### Summary

The certificate of the update center expires soon (31st of May 2022) and must be renewed for 1 year.
It is used for updates of the update-center JSON and tools metadata ("crawler").

https://github.com/jenkins-infra/runbooks/tree/main/updates#renew-update-center-certificates

Before closing this issue, make sure to set a calendar event in the infra team calendar 1 month before

Note: 2022 renewal: https://github.com/jenkins-infra/helpdesk/issues/2950

**Comment by smerle33** [comment]
@olblak been contacted by @dduportal to sign the certificate with the CA key.

**Comment by dduportal** [comment]
Been in contact with @olblak today. He now has my public GPG key and will send me later the certificate (or the CA key) to unblock.

**Comment by dduportal** [comment]
Update: sent an email to the board to ask the permission of having access to the CA private key.

**Comment by NotMyFault** [comment]
> Update: sent an email to the board to ask the permission of having access to the CA private key.

I included the topic in the agenda for the next governance meeting, although the 1st of May is a bank holiday in Germany, so we may postpone it to the next meeting or work our way out in async communication.

Thanks!

**Comment by dduportal** [comment]
> > Update: sent an email to the board to ask the permission of having access to the CA private key.
> 
> I included the topic in the agenda for the next governance meeting, although the 1st of May is a bank holiday in Germany, so we may postpone it to the next meeting or work our way out in async communication.
> 
> Thanks!

cc @olblak for info given the discussion about the CA key could be postponed, do you mind generated (one last time) the Update Center certificate for 6 months (instead of 1 year) and send it (encrypted) to us?

**Comment by MarkEWaite** [comment]
> > > Update: sent an email to the board to ask the permission of having access to the CA private key.
> > 
> > 
> > I included the topic in the agenda for the next governance meeting, although the 1st of May is a bank holiday in Germany, so we may postpone it to the next meeting or work our way out in async communication.
> > Thanks!

3 of 5 governance board members approved the request that Damien be given access to the key.

**Comment by dduportal** [comment]
- We reached the "30 days before certificate expiration" causing the builds of the jenkins-infra/update-center2 (on trusted.ci) to fail since yesterday as per the default `-DCERTIFICATE_MINIMUM_VALID_DAYS="${CERTIFICATE_MINIMUM_VALID_DAYS:-30}"` flag at https://github.com/jenkins-infra/update-center2/blob/master/site/generate.sh#L70
  - Changed the freestyle job configuration to `RECENT_RELEASES_MAX_AGE_HOURS=48 CERTIFICATE_MINIMUM_VALID_DAYS=14 <...> ./site/publish.sh` (added `CERTIFICATE_MINIMUM_VALID_DAYS=14` to unblock and setting `RECENT_RELEASES_MAX_AGE_HOURS` to `48 instead of `6` to catch up missing plugin releases)

- Checked with @olblak : I should received the CA key soon as validated by the board, to allow me signing the CSR prepared by @smerle33

**Comment by dduportal** [comment]
Update:

- The jenkins-infra/update-center2 builds are green again. Published plugins backlog should have been treated: setting it back to `6` hours max (to avoid too long builds).
- The crawler build fails during publication as no HTML/JSON assets were published so non published. **However**, the real error resides in the builds logs:
```
Caught: java.security.cert.CertificateExpiredException: NotAfter: Thu Jun 01 07:13:16 UTC 2023
18:21:38  java.security.cert.CertificateExpiredException: NotAfter: Thu Jun 01 07:13:16 UTC 2023
18:21:38  	at org.jvnet.hudson.update_center.Signer.getCertificateChain(Signer.java:211)
18:21:38  	at org.jvnet.hudson.update_center.Signer.sign(Signer.java:106)
18:21:38  	at org.jvnet.hudson.update_center.Signer$sign$0.call(Unknown Source)
18:21:38  	at lib.DataWriter.write(DataWriter.groovy:20)
18:21:38  	at lib.DataWriter$write.call(Unknown Source)
18:21:38  	at terraform.run(terraform.groovy:20)
18:21:38  	at runner$_run_closure1.doCall(runner.groovy:13)
18:21:38  	at runner.run(runner.groovy:10)
```

**Comment by dduportal** [comment]
Update: 

- Both builds were successfull
- Update center build updated to remove the 14 days validity (to ensure eveyrthing is clean)

**Comment by dduportal** [comment]
Update:

- Everything looks good: update center job runs well with the default "30 days" validity, crawler is good
- Updated plugins index on ci.jenkins.io: new plugins found, proving it's working
- Updated the description of the 2 credentials with the year to avoid mixing it up in trusted.ci.jenkins.io
- Reminder to remove the 2022 credential in 1 week 


=> we can close the issue.

Thanks @smerle33 @olblak !
